protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java

```diff
@@ -241,23 +241,21 @@ private void addFlagsResponses(ImapSession session, SelectedMailbox selected, Im
     protected void addFlagsResponses(ImapSession session, SelectedMailbox selected, ImapProcessor.Responder responder, boolean useUid, MessageRange messageSet, MessageManager mailbox, MailboxSession mailboxSession) throws MailboxException {
 
         final MessageResultIterator it = mailbox.getMessages(messageSet, FetchGroupImpl.MINIMAL,  mailboxSession);
+        final boolean qresyncEnabled = EnableProcessor.getEnabledCapabilities(session).contains(ImapConstants.SUPPORTS_QRESYNC);
+        final boolean condstoreEnabled = EnableProcessor.getEnabledCapabilities(session).contains(ImapConstants.SUPPORTS_CONDSTORE);
+        final boolean isModSeqPermanent =  mailbox.getMetaData(false, mailboxSession, FetchGroup.NO_COUNT).isModSeqPermanent();
```

Comment:
`MessageManager` `boolean isModSeqPermanent(MailboxSession session);` would be even a better call for that?

Reply:
It's deprecated on the StoreMessageManager and does not exist in MessageManager but in it's inner Metadata class.

Reply:
It's not in the API and that's what we care about no?

Actually we see here that this could avoid executing some remote queries on Cassandra for a method that is returning always true. Last Uid and highest modseq are also retrieved by that call.